modules/core/entity/modules/fields/farm_entity_fields.base_fields.inc

```diff
@@ -99,6 +99,64 @@ function farm_entity_fields_log_base_fields() {
   return $fields;
 }
 
+/**
+ * Define common organization base fields.
+ */
+function farm_entity_fields_organization_base_fields() {
+  $field_info = [
+    'data' => [
+      'type' => 'string_long',
+      'label' => t('Data'),
+      'hidden' => TRUE,
+    ],
+    'file' => [
+      'type' => 'file',
+      'label' => t('Files'),
+      'file_directory' => 'farm/organization/[date:custom:Y]-[date:custom:m]',
+      'multiple' => TRUE,
+      'weight' => [
+        'form' => 90,
+        'view' => 90,
+      ],
+    ],
+    'image' => [
+      'type' => 'image',
+      'label' => t('Images'),
+      'file_directory' => 'farm/organization/[date:custom:Y]-[date:custom:m]',
+      'multiple' => TRUE,
+      'weight' => [
+        'form' => 89,
+        'view' => 89,
+      ],
+    ],
+    'notes' => [
+      'type' => 'text_long',
+      'label' => t('Notes'),
+      'weight' => [
+        'form' => 95,
+        'view' => 10,
+      ],
+    ],
+    'user' => [
```

Comment:
Maybe we could remove the user field to leverage it from a Group integration instead?

Reply:
This is a good question @pcambra! I've been thinking about it ever since it came up in conversation.

(For readers who may not be familiar with recent conversations, this is referring to ideas for a potential contrib farmOS module that uses the data types provided by this PR + the Drupal Group module to enable a "multi-tenant" farmOS instance with access control between groups/farms).

In the context of the multi-tenant farmOS module, there needs to be a way to link users to Groups (farms), so that access control can be applied to them. The Group module provides its own way of doing this.

However, as of right now the plan is for that to be a contrib module which extends core farmOS data model. So perhaps the question right now is: are there use-cases for linking users to farms outside of the multi-tentant access controlled context? If so, then we may need a user relationship in core as well.

In reviewing the forum discussion, I found this, which I had forgotten about:

> Discussed on the farmOS dev call today (with @Symbioquine @paul121 @Fosten):
> - We decided to wait on adding the “Users” field to Organization entities. There isn’t a direct use for it in the current PR, and we may want to consider putting an Organization reference field on the User records instead.

Since there *is* a use-case now (multi-tenant farmOS), perhaps we should discuss it together again. If there is any possibility that we will want a user relationship in the *core* data model, then we should add that now and the multi-tenant module should work around it IMO. Otherwise we may end up with multiple different implementations for the same thing.

Reply:
I copied most of this to the forum topic: https://farmos.discourse.group/t/organization-level-data/2003/10

Let's discuss over there (and in a future dev call).